internal/controllers/machine/machine_controller_phases.go

```diff
@@ -174,6 +174,16 @@ func (r *Reconciler) reconcileExternal(ctx context.Context, cluster *clusterv1.C
 // reconcileBootstrap reconciles the Spec.Bootstrap.ConfigRef object on a Machine.
 func (r *Reconciler) reconcileBootstrap(ctx context.Context, cluster *clusterv1.Cluster, m *clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+	var externalResult external.ReconcileOutput
+	var err error
+
+	// Call generic external reconciler if we have an external reference.
+	if m.Spec.Bootstrap.ConfigRef != nil {
```

Comment:
We end up checking this twice - once here and its inverse in L196 which isn't an ideal flow, but I'm not sure what the alternative is.